### PR TITLE
SYCL Resource

### PR DESCRIPTION
**Pull request by homerdin**
Initial PR to add a SYCL resource.

### CODE REVIEW

File: include/camp/defines.hpp [hunk 1]
```diff
@@ -88,6 +88,13 @@ namespace camp
 
 #define CAMP_SUPPRESS_HD_WARN
 
+#elif defined(SYCL_LANGUAGE_VERSION)
+#define CAMP_HAVE_SYCL 1
+#define CAMP_DEVICE
+#define CAMP_HOST_DEVICE
+#pragma warn("Sycl compiler!")
```
Comment (trws) [inline include/camp/defines.hpp:95]: Do you mean to leave this in?

**Review by trws — COMMENTED** [review]
This looks nice and clean, everything seems to be set up the way it should be too.  Just want to know if you intend that warning to stay, I'd probably say remove it, otherwise LGTM.

File: include/camp/defines.hpp (hunk 1 above)
Comment (homerdin) [reply include/camp/defines.hpp]: No, I'd forgotten to remove it.  I'll remove it now.

**Comment by homerdin** [comment]
@trws I've updated the resource to work through/return a pointer to the sycl::queue.  Returning copies can cause issues with the memory associated with the queue.  

I've also updated the variable names, "stream" -> "qu"

**Comment by trws** [comment]
Ok, this looks pretty good.  I need to figure out what's going on with appveyor though, almost looks like a cmake issue in the newer version over there.

**Comment by trws** [comment]
Found the appveyor issue.  I'm going to merge this based on the results for linux, since the appveyor results are unrelated.  Thanks!

**Pull request merged**